packages/experimental-utils/src/ast-utils/predicates.ts

```diff
@@ -211,7 +211,7 @@ function isAwaitExpression(
 function isAwaitKeyword(
   node: TSESTree.Token | TSESTree.Comment | undefined | null,
 ): node is TSESTree.KeywordToken & { value: 'await' } {
-  return node?.type === AST_TOKEN_TYPES.Identifier && node.value === 'await';
+  return node?.type === AST_TOKEN_TYPES.Keyword && node.value === 'await';
```

Comment:
the logic is correct - the type is incorrect.
due to limitations in eslint's default parser - they emit `await` as an identifier, so we do as well